example/server/integrations.py

```diff
@@ -970,3 +988,161 @@ def info_integration(request: Request, asset: Asset, lang: str):
             }
         },
     }
+
+
+def check_delivery_method(
+    asset: str,
+    list_delivery_methods_func: Callable[[OffChainAsset], List],
+    delivery_method: str,
+) -> bool:
+    if delivery_method:
+        bdms = list_delivery_methods_func(get_offchain_asset(asset))
+        if not any(bdm.name == delivery_method for bdm in bdms):
+            # This method is not supported
+            return False
+    return True
+
+
+def check_country_code(asset: str, country_code: str) -> bool:
+    if country_code and country_code not in get_offchain_asset(
+        asset
+    ).country_codes.split(","):
+        # The contry code is not supported by the anchor.
+        return False
+    return True
+
+
+class MySEP38AnchorIntegration(SEP38AnchorIntegration):
+    def get_prices(
+        self,
+        sell_asset: str,
+        sell_amount: str,
+        sell_delivery_method: str = None,
+        buy_delivery_method: str = None,
+        country_code: str = None,
+    ) -> List[GetPricesResponse]:
+        exchange_pairs = list_exchange_pairs(buy_asset=sell_asset)
+
+        prices = []
+        for exchange_pair in exchange_pairs:
+            if is_stellar_asset(sell_asset):
+                # We are selling a stellar asset. Buying asset is offchain.
+                if not check_delivery_method(
+                    asset=exchange_pair.sell_asset,
+                    list_delivery_methods_func=get_buy_delivery_methods,
```

Comment:
I'm not a fan of passing functions to other functions in Python, can we write these utility functions in a way that doesn't require this?

Reply:
Maybe instead of the `asset` parameter we can have `sell_asset` and `buy_asset` so the `check_delivery_method()` function knows which `get_*_delivery_methods()` function to call.